nemo/collections/vlm/mllama/model/base.py

```diff
@@ -467,14 +469,23 @@ def forward(
                 total_len=position_ids.shape[1],
             )
 
+            original_position_id = position_ids[0]
+
+            if inference_params is not None:
+                inference_params.xattn_caches = xattn_caches
+                inference_params.cross_attention_masks = cross_attention_masks
+                inference_params.full_text_row_masked_out_mask = full_text_row_masked_out_mask
+        else:
+            original_position_id = [cross_attention_masks.shape[2] - 1]
```

Comment:
is this a single id for generation? i feel original position id name is a bit confusing